File: api/anki_api.js
class Client {
  constructor(address = "127.0.0.1", port = 5555, token = "") {
      this.port = port;
      this.address = address;
      this.base_url = `http://${this.address}:${this.port}/anki_api`;
      this.token = token;
  }

  async create_car(address) {
      if (this.token === "") {
          const response = await fetch(`${this.base_url}/create_car?address=${address}`);
          this.token = await response.text();
          return this.token;
      } else {
          return this.token;
      }
  }

  async set_speed(speed, accel = 1000) {
      const response = await fetch(`${this.base_url}/set_speed?token=${this.token}`, {
          method: 'POST',
          headers: {
              'Content-Type': 'application/json'
          },
          body: JSON.stringify({
              speed: speed,
              accel: accel
          })
      });
      return await response.text();
  }

  async left_lane(lanes = 1.0) {
      const response = await fetch(`${this.base_url}/left_lane?token=${this.token}`, {
          method: 'POST',
          headers: {
              'Content-Type': 'application/json'
          },
          body: JSON.stringify({
              lanes: lanes
          })
      });
      return await response.text();
  }

  async right_lane(lanes = 1.0) {
      const response = await fetch(`${this.base_url}/right_lane?token=${this.token}`, {
          method: 'POST',
          headers: {
              'Content-Type': 'application/json'
          },
          body: JSON.stringify({
              lanes: lanes
          })
      });
      return await response.text();
  }

  async get_cars(active = false) {
      const response = await fetch(`${this.base_url}/get_cars`, {
          method: 'POST',
          headers: {
              'Content-Type': 'application/json'
          },
          body: JSON.stringify({
              active: active
          })
      });
      const data = await response.json();
      return data.cars;
  }

  async disconnect(force_stop = true) {
      const response = await fetch(`${this.base_url}/disconnect?token=${this.token}`, {
          method: 'POST',
          headers: {
              'Content-Type': 'application/json'
          },
          body: JSON.stringify({
              force_stop: force_stop
          })
      });
      return await response.text();
  }
}

File: api/anki_api.py
import requests

class Client:
    def __init__(self, address: str = "127.0.0.1", port: int = 5555, token: str = ""):
        """Init the API

        Args:
            address (str): Local address
            port (int, optional): Local port. Defaults to 5555.
        """
        self.port: int = port
        self.address: str = address
        self.base_url = f"http://{self.address}:{self.port}/anki_api"
        self.token: str = token
    
    def create_car(self, address: str):
        if self.token == "":
            r = requests.post(f"{self.base_url}/create_car", params={
                "address": address
            })
            self.token = r.text
            return self.token
        else:
            return self.token
    
    def set_speed(self, speed: int, accel: int = 1000):
        r = requests.post(f"{self.base_url}/set_speed", params={
            "token": self.token
        }, json={
            "speed": speed,
            "accel": accel
        })
        return r.text
    
    def left_lane(self, lanes: float = 1.0):
        r = requests.post(f"{self.base_url}/left_lane", params={
            "token": self.token,
        }, json={
            "lanes": lanes
        })
        return r.text
        
    def right_lane(self, lanes: float = 1.0):
        r = requests.post(f"{self.base_url}/right_lane", params={
            "token": self.token,
        }, json={
            "lanes": lanes
        })
        return r.text

    def get_cars(self, active: bool = False):
        r = requests.post(f"{self.base_url}/get_cars", json={
            "active": active
        })
        return r.json()["cars"]

    def disconnect(self, force_stop: bool = True):
        r = requests.post(f"{self.base_url}/disconnect", json={
            "force_stop": force_stop
        }, params={
            "token": self.token
        })
        return r.text